include/zephyr/net/wifi_mgmt.h

```diff
@@ -229,6 +236,14 @@ enum net_event_wifi_cmd {
 #define NET_EVENT_WIFI_AP_STA_DISCONNECTED			\
 	(_NET_WIFI_EVENT | NET_EVENT_WIFI_CMD_AP_STA_DISCONNECTED)
 
+/** Wi-Fi version */
+struct wifi_version {
+	/** Driver version */
+	char drv_version[WIFI_DRV_VERSION_MAX_LEN];
+	/** Firmware version */
+	char fw_version[WIFI_FW_VERSION_MAX_LEN];
+};
```

Comment:
Normally I would do it like this but this data structure is quite large and uses lot of stack. The data is probably mostly static and could be stored into ROM, and even if not so, the driver could use static char array to store the data only once. So I think we could safely use a pointer here as you originally had. Perhaps `const char * const drv_version` and similar for the `fw_version`

Reply:
Sure, but then we need to add checks to make sure point is still valid.

Reply:
> Sure, but then we need to add checks to make sure point is still valid.

If the information is stored somewhere in ROM or in static array, then we can trust the pointer to be a valid one. It would be madness in the driver to store this kind of non changing data to a stack variable.

Reply:
Absolutely, but you never know :).

Reply:
We can certainly document how the data pointed by the pointer should be stored. If the user wants to shoot itself to the foot by doing something weird, then go ahead and best luck :wink:

Reply:
I agree with @jukkar to keep as pointers only, as these would be const ROM locations and will never change.